From johnpol@2ka.mipt.ru Thu Oct  5 03:26:12 2006
Date: Thu, 5 Oct 2006 14:26:02 +0400
From: Evgeniy Polyakov <johnpol@2ka.mipt.ru>
To: GregKH <greg@kroah.com>
Subject: W1: ioremap balanced with iounmap
Message-ID: <20061005102601.GG1015@2ka.mipt.ru>
Content-Disposition: inline

From: Amol Lad <amol@verismonetworks.com>

ioremap must be balanced with iounmap in error path.

Please consider for 2.6.19.

Signed-off-by: Amol Lad <amol@verismonetworks.com>
Signed-off-by: Evgeniy Polyakov <johnpol@2ka.mipt.ru>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/w1/masters/matrox_w1.c |    2 ++
 1 file changed, 2 insertions(+)

--- gregkh-2.6.orig/drivers/w1/masters/matrox_w1.c
+++ gregkh-2.6/drivers/w1/masters/matrox_w1.c
@@ -215,6 +215,8 @@ static int __devinit matrox_w1_probe(str
 	return 0;
 
 err_out_free_device:
+	if (dev->virt_addr)
+		iounmap(dev->virt_addr);
 	kfree(dev);
 
 	return err;
